refactor: Do not use the Symfony validator to validate the honeypot captcha

aragon999 opened the pull request:
### 1. Why is this change necessary?
By now the Symfony validator also supports attributes instead of an explicit declaration in a dedicated method.

### 2. What does this change do, exactly?
Use the attribute, reduce loc.

### 3. Describe each step to reproduce the issue or behaviour.
:shrug:

### 4. Please link to the relevant issues (if any).
\-

### 5. Checklist

- [ ] I have written tests and verified that they fail without my change
- [ ] I have created a [changelog file](https://github.com/shopware/shopware/blob/trunk/adr/2020-08-03-implement-new-changelog.md) with all necessary information about my changes
- [ ] I have written or adjusted the documentation according to my changes
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfilled them

aragon999 committed:
refactor: Use attribute instead of declaration

aragon999 commented on src/Storefront/Framework/Captcha/HoneypotCaptcha.php:
```diff
@@ -23,14 +23,6 @@ public function __construct(private readonly ValidatorInterface $validator)
-    /**
-     * Default method for determining constraints when using the Symfony validator.
-     */
-    public static function loadValidatorMetadata(ClassMetadata $metadata): void
```
Mh, due to the failed BC checker, should I deprecate the method? And remove its content? 🤔

mitelg replied:
yes, we should deprecated it. but maybe do nothing if 6.8 feature is active? :thinking: would otherwise breaking usages

aragon999 committed:
fixup: Remove Symfony validator of the honeypot captcha completly

aragon999 commented:
@mitelg I have come back to this PR and I think in general the Symfony validator should not be used at all.

On the one hand it is too much overhead, for a single string comparison.

Furthermore it might even introduce "bad practice", as one might try to inherit the original `HoneypotCaptcha` class, to modify the behavior (however the constructor is internal), so the decoration pattern should be used.

I changed the PR accordingly and adjusted the changelog, what do you think?

mitelg commented on src/Storefront/Framework/Captcha/HoneypotCaptcha.php:
```diff
@@ -36,9 +44,13 @@ public static function loadValidatorMetadata(ClassMetadata $metadata): void
+        if (!Feature::isActive('v6.8.0.0')) {
+            $this->honeypotValue = $request->get(self::CAPTCHA_REQUEST_PARAMETER, '');
+
+            return \count($this->validator->validate($this)) < 1;
```
the validator dep could also be deprecated

aragon999 replied:
You mean in the constructor? But the constructor is internal, or do you mean when "resolving" the deprecations?

mitelg replied:
yes, so it is not just forgotten, if this part is removed :wink: but also PHPStan might throw an error, if there is a only written property :thinking:

aragon999 replied:
Yes, phpstan will complain :-) 
<img width="1262" height="185" alt="image" src="https://github.com/user-attachments/assets/e31f8a88-b5eb-4d07-a9bf-b93b6232694a" />

So I think this is a not needed deprecation :-)

mitelg commented:
@aragon999 makes sense :+1:

mitelg committed:
Merge branch 'trunk' into refactor/attribute-instead-of-method

mitelg commented:
thanks for your contribution @aragon999 👍🎉💙
